src/screens/TopicPageScreen/index.js

```diff
@@ -67,7 +67,9 @@ const TopicPageScreen = (props) => {
   ).current;
 
   const [feedsContext, dispatch] = React.useContext(Context).feeds;
-  const feeds = feedsContext.topicFeeds;
+  const feeds = feedsContext.topicFeeds
+    ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(topicName))
+    : [];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The filtering operation on `feedsContext.topicFeeds` could potentially be expensive if the number of feeds is large. Consider optimizing this by using a data structure that allows for faster lookup, such as a Set or Map, for the topics in each feed.
```diff
- const feeds = feedsContext.topicFeeds
-   ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(topicName))
-   : [];
+ const feeds = feedsContext.topicFeeds
+   ? feedsContext.topicFeeds.filter((feed) => {
+       const topicSet = new Set(feed?.topics);
+       return topicSet.has(topicName);
+     })
+   : [];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->